config/config.go

```diff
@@ -95,9 +97,18 @@ func WithOpenTelemetryConfiguration(cfg OpenTelemetryConfiguration) Configuratio
 	})
 }
 
+// ParseJSON parses a JSON configuration file into an OpenTelemetryConfiguration.
+func ParseJSON(r io.Reader) (*OpenTelemetryConfiguration, error) {
+	var cfg OpenTelemetryConfiguration
+	decoder := json.NewDecoder(r)
+	if err := decoder.Decode(&cfg); err != nil {
```

Comment:
Can you make some more advanced test? I am not sure if it works "out of the box". We may be missing JSON tags. See: https://github.com/open-telemetry/opentelemetry-go-contrib/blob/ba19074a6785b2b65c86a52c0649cf040319fa96/Makefile#L322

@codeboten Am I correct?

Reply:
@pellared / @codeboten, I wasn't super clear on what the expected input would be, so perhaps I've overlooked something. I'd thought the `OpenTelemetryConfiguration` was specifically designed to build the parameters as `mapstructures` so the user could choose to define their config using either yaml or json. I (possibly naively) was building this to accept any valid json that had the [required property](https://github.com/open-telemetry/opentelemetry-configuration/blob/62138088f545154d15aca4e74fe6f6ae93c94454/schema/opentelemetry_configuration.json#L33) from the configuration input used in the Makefile:
https://github.com/open-telemetry/opentelemetry-go-contrib/blob/ba19074a6785b2b65c86a52c0649cf040319fa96/Makefile#L324
 Perhaps we can clarify what are the expected valid inputs for this (and #4373), and what we'd expect as output on valid/invalid cases? I feel this would help with implementation and writing more comprehensive tests. lmkwyt!

Reply:
I was thinking of removing this line: 

https://github.com/open-telemetry/opentelemetry-go-contrib/blob/ba19074a6785b2b65c86a52c0649cf040319fa96/Makefile#L322

As then the `OpenTelemetryConfiguration` type will have  `mapstructure`, `json` and `yaml` tags. See: 

https://github.com/omissis/go-jsonschema/blob/6fcc83b8eebe874eeebc20474775280877cc62fe/main.go#L167-L168

Then we could simply provide examples how one can parse input to `OpenTelemetryConfiguration` using different libraries.

The parsing method do NOT need to do any validation e.g. check for required fields. The validation has to be done by `NewSDK` because the user could even create `OpenTelemetryConfiguration`  "by hand" without parsing any JSON or YAML.